sdk/src/main/java/com/google/cloud/dataflow/sdk/io/OffsetBasedSource.java

```diff
@@ -264,7 +264,18 @@ public final boolean start() throws IOException {
 
     @Override
     public final boolean advance() throws IOException {
-      return advanceImpl() && rangeTracker.tryReturnRecordAt(isAtSplitPoint(), getCurrentOffset());
+      if (!advanceImpl()) {
+        // Reader could not read another record. Add an extra split point in order to "finish" the
```

Comment:
Maybe create a method reportEOF() that would be implemented as tryReturnRecordAt() ? I'd prefer if at least the calling code did not conflate the concepts of EOF and returning a record.